packages/common/src/logging/config.ts

```diff
@@ -0,0 +1,358 @@
+  sinks?: Array<(entry: LogEntry) => void>;
+  /** Disable telemetry headers in requests. Defaults to false. */
+  disableTelemetryHeaders?: boolean;
+  /** Array of exact namespace strings to allow. If undefined or empty, all namespaces are enabled. Example: ["aa-infra", "wallet-apis"] */
+  enabledNamespaces?: string[];
+};
+
+/**
+ * Base context attached to all log entries from a logger instance.
+ */
+export type LoggerContextBase = {
+  /** Package name (e.g., "@alchemy/aa-infra") */
+  package: string;
+  /** Package version (e.g., "1.0.0") */
+  version: string;
+};
+
+/**
+ * A single log entry emitted by the diagnostics logger.
+ */
+export type LogEntry = {
+  /** Timestamp in milliseconds since epoch (Date.now()) */
+  ts: number;
+  /** Log level of this entry */
+  level: LogLevel;
+  /** Optional namespace for filtering/grouping (e.g., "aa-infra", "wallet-apis") */
+  namespace: string | undefined;
+  /** The log message */
+  message: string;
+  /** Optional structured data attached to this log entry */
+  data?: Record<string, unknown>;
+  /** Package context (name and version) */
+  context: LoggerContextBase;
+};
+
+let globalConfig: Required<DiagnosticsConfig> | undefined;
+
+function defaultConfig(): Required<DiagnosticsConfig> {
+  const env = (typeof process !== "undefined" && process.env) || {};
+
+  // Default level: INFO in dev, ERROR in prod
+  const isDev = env.NODE_ENV === "development";
+  const defaultLevel = isDev ? LogLevel.INFO : LogLevel.ERROR;
+
+  let level = defaultLevel;
+  const envLevel = env.ALCHEMY_LOG_LEVEL;
+  if (envLevel) {
+    const map: Record<string, LogLevel> = {
+      error: LogLevel.ERROR,
+      warn: LogLevel.WARN,
+      info: LogLevel.INFO,
+      debug: LogLevel.DEBUG,
+      verbose: LogLevel.VERBOSE,
+    };
+    const normalized = envLevel.toLowerCase();
+    if (normalized in map) {
+      level = map[normalized];
+    } else {
+      const asNum = Number(envLevel);
+      if (!Number.isNaN(asNum) && asNum >= 0 && asNum <= 4)
+        level = asNum as LogLevel;
+    }
+  }
+
+  return {
+    level,
+    redact: {
+      keys: (key: string) =>
+        /^(authorization|apiKey|jwt|privateKey|secret|password)$/i.test(key),
+      replacer: () => "[REDACTED]",
+    },
+    sinks: [consoleSink],
+    disableTelemetryHeaders: false,
+    enabledNamespaces: [],
+  };
+}
+
+/**
+ * Gets the current global logger configuration.
+ * If not yet initialized, returns default configuration based on environment.
+ *
+ * @returns {Required<DiagnosticsConfig>} The current global configuration with all fields populated
+ * @example
+ * ```ts
+ * import { getGlobalLoggerConfig } from "@alchemy/common";
+ *
+ * const config = getGlobalLoggerConfig();
+ * console.log("Current log level:", config.level);
+ * ```
+ */
+export function getGlobalLoggerConfig(): Required<DiagnosticsConfig> {
+  if (!globalConfig) {
+    globalConfig = defaultConfig();
+  }
```

Comment:
shorthand for this, if you want to use it
```suggestion
  globalConfig ??= defaultConfig();
```